### PR TITLE
Dynamically adjust StatsTable height

### PR DESCRIPTION
**Pull request by itanasi**
Dynamically set the StatsTable height based on whatever is in the lower right corner
image
image
image

### CODE REVIEW

File: core/src/com/unciv/ui/cityscreen/CityStatsTable.kt [hunk 1]
```diff
@@ -34,10 +36,11 @@ class CityStatsTable(val cityScreen: CityScreen): Table() {
         outerPane = ScrollPane(innerTable)
         outerPane.setOverscroll(false, false)
         outerPane.setScrollingDisabled(true, false)
-        add(outerPane).maxHeight(cityScreen.stage.height / 2)
+        statsCell = add(outerPane)
+        statsCell.maxHeight(cityScreen.stage.height / 2)
```
Comment (SomeTroglodyte) [inline core/src/com/unciv/ui/cityscreen/CityStatsTable.kt:40]: Likely redundant

File: core/src/com/unciv/ui/cityscreen/CityStatsTable.kt [hunk 2]
```diff
@@ -34,10 +36,11 @@ class CityStatsTable(val cityScreen: CityScreen): Table() {
...
     }
 
-    fun update() {
+    fun update(height: Float = cityScreen.stage.height / 2) {
```
Comment (SomeTroglodyte) [inline core/src/com/unciv/ui/cityscreen/CityStatsTable.kt:43]: default really used??

**Comment by SomeTroglodyte** [comment]
> [redo Gradle?](https://github.com/yairm210/Unciv/pull/6990/commits/8d6c3232c4eb85b33979765378e3e20ea3415fb7)

A non-collaborator can't go to the failed run and repeat it directly? Show checks - both on the left on the individual run and the top right:
image
... needs elevated permissions? Or not? Other people close and reopen or go draft-ready but I think if any contributor can, then we should wiki the cleaner way - prettier for history, not how to get there...

File: core/src/com/unciv/ui/cityscreen/CityStatsTable.kt (hunk 2 above)
Comment (itanasi) [reply core/src/com/unciv/ui/cityscreen/CityStatsTable.kt:43]: No, just for safety

**Comment by itanasi** [comment]
Nope, couldn't find the rerun button. Draft and back may work instead

File: core/src/com/unciv/ui/cityscreen/CityStatsTable.kt (hunk 1 above)
Comment (itanasi) [reply core/src/com/unciv/ui/cityscreen/CityStatsTable.kt:40]: You mean setting maxHeight during init()? Are we guaranteed update()?

Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/cityscreen/CityStatsTable.kt:40]: Exactly - are we?

**Review by SomeTroglodyte — APPROVED** [review]
Looks like it will work just fine.

File: core/src/com/unciv/ui/cityscreen/CityStatsTable.kt (hunk 1 above)
Comment (itanasi) [reply core/src/com/unciv/ui/cityscreen/CityStatsTable.kt:40]: Functionally my guess is yes to populate the tables with values. But unsure. Likely safe to remove

Comment (SomeTroglodyte) [reply core/src/com/unciv/ui/cityscreen/CityStatsTable.kt]: If after analysis or debugging you're still unsure, leave it. If you're sure, ditch. Me, I just like to prod a little.

**Comment by Ouaz** [comment]
Hello, can I make a request?

Could it possible to keep the stats lines always displayed, even when scrolling inside the table?

Because currently, when scrolling, the first stats line is hidden, and when modifying focus or specialists assignment, it's not possible to see at the same time what it changes in the stats line (it requires to scroll back up each time). With the stats lines always displayed, it will be more convenient to test different focus/specialists settings.

stats_scroll

**Comment by SomeTroglodyte** [comment]
> make a request

Nice gimp/photoshop/whatever work. Thought so too earlier but forgot too quickly to comment.

**Comment by itanasi** [comment]
I think that's doable

**Comment by itanasi** [comment]
Behold
image

**Pull request merged**